src/openforms/forms/tests/e2e_tests/test_service_fetch.py

```diff
@@ -0,0 +1,587 @@
+from playwright.async_api import Page, expect
+from zgw_consumers.constants import APITypes, AuthTypes
+from zgw_consumers.models import Service
+
+from openforms.config.models import GlobalConfiguration
+from openforms.tests.e2e.base import E2ETestCase, browser_page, create_superuser
+from openforms.variables.constants import DataMappingTypes, FormVariableSources
+from openforms.variables.models import ServiceFetchConfiguration
+from openforms.variables.tests.factories import ServiceFetchConfigurationFactory
+
+from ..factories import FormFactory, FormVariableFactory
+
+
+@contextmanager
+def phase(desc: str):
+    yield
+
+
+async def add_new_variable_with_service_fetch(page: Page):
+    await page.get_by_role("tab", name="Logic").click()
+
+    await page.get_by_text("Add rule").click()
+    await page.get_by_text("Simple").click()
+
+    await page.locator('select[name="variable"]').select_option(
+        label="Environment (environment)"
+    )
+    await page.locator('select[name="operator"]').select_option(label="is equal to")
+    await page.locator('select[name="operandType"]').select_option(label="value")
+    await page.locator('input[name="operand"]').fill("foo")
+
+    await page.get_by_text("Add action").click()
+    await page.locator('select[name="action.type"]').select_option(
+        label="fetch the value for a variable from a service"
+    )
+    # FIXME Name conflict with the select above
+    await page.locator('select[name="variable"]').last.select_option(
+        label="Variable 1 (variable1)"
+    )
+
+    await expect(
+        page.get_by_text("Fetch configuration: (not configured yet)")
+    ).to_have_count(1)
+
+
+async def fill_in_service_fetch_form(page: Page, data: dict, save_text: str = "Save"):
+    await page.get_by_label("Name").fill(data["name"])
+    await page.get_by_label("HTTP method").select_option(data["method"])
+    await page.get_by_label("Service").select_option(label=data["service"])
+    await page.get_by_label("Path").fill(data["path"])
+
+    if not await page.locator(
+        'div.form-row.field-queryParams input[name="key"]'
+    ).count():
+        await page.locator("div.form-row.field-queryParams span.addlink").click()
+    await page.locator('div.form-row.field-queryParams input[name="key"]').fill(
+        list(data["query_params"].items())[0][0]
+    )
+    await page.locator('input[name="value-0"]').fill(
+        list(data["query_params"].items())[0][1][0]
+    )
+
+    if not await page.locator('div.form-row.field-headers input[name="key"]').count():
+        await page.locator("div.form-row.field-headers span.addlink").click()
+    await page.locator('div.form-row.field-headers input[name="key"]').last.fill(
+        list(data["headers"].items())[0][0]
+    )
+    await page.locator('input[name="value"]').fill(list(data["headers"].items())[0][1])
+
+    # FIXME doesn't seem to work with `get_by_label`
+    await page.locator('textarea[name="body"]').fill(data["request_body"])
```

Comment:
I think this works via the HTML `id` attribute on the textarea and the label `for` attribute. The textarea is missing the ID attribute.